### ISSUE TITLE
ApolloClient.fetch() does not make a request to the graphQL server.

### ISSUE THREAD

**Issue by MaruthiBasava** [apollographql/apollo-ios#871]
```swift 

func fetch<Query: GraphQLQuery>(query: Query,
                                    cachePolicy: CachePolicy = CachePolicy.returnCacheDataElseFetch,
                                    queue: DispatchQueue = DispatchQueue.main) -> Signal<Query.Data, Error> {
        return Signal { [weak self] observer in
            
            
            let cancellable = self?.apollo.fetch(query: query, cachePolicy: cachePolicy, queue: queue) { result in
                
                switch result {
                case .success(let res):
                    if let data = res.data {
                        
                        observer.receive(data)
                    } else if let errors = res.errors {
                        
                        print("**", errors)
                    }
                case .failure(let err):
                    
                    observer.receive(completion: .failure(err))
                }
                
            }
            
            return BlockDisposable {
                cancellable?.cancel()
            }
            
        }
        
    }
```

the Inside of the fetch's completion block never gets called nor does my graphql server log that client has requested. No error messages are logged in Xcode. I spent a lot of time debugging, running out of options. I tried multiple GraphQL endpoints to make sure the fault isn't within mine.

**Comment by designatednerd** [comment]
Hi - I'm not familiar with the `Signal` wrapper you're using. What library is that coming from? 

It also might be worth throwing in a 

```swift
guard let self = self else {
  print("Self was deallocated")
  return 
}
```

at the top of the closure creating the `Signal` wrapper to see if that's the issue - that would indicate that you're not holding on to the returned `Signal` and it's being deallocated before it can be called.

**Comment by MaruthiBasava** [comment]
You solved my issue! My client class was being deallocated so the fetches were never happening.

**Comment by designatednerd** [comment]
Yay! Yeah [we fixed a couple of retain cycles in `0.18.0`](https://github.com/apollographql/apollo-ios/blob/master/CHANGELOG.md#v0180), so that might have had something to do with it 🙃
